File: CarCareHub_/CarCareHub.Services/Database/Korpa.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CarCareHub.Services.Database
{
    public class Korpa
    {
        public int KorpaId { get; set; }
        public virtual ICollection<NarudzbaStavka> NarudzbaStavkas { get; set; } = new List<NarudzbaStavka>();
        public bool Vidljivo { get; set; }
        public int? ProizvodId { get; set; }
        public virtual Proizvod? Proizvod { get; set; }
        public int? KlijentId { get; set; }
        public virtual Klijent? Klijent { get; set; }
        public int? AutoservisId { get; set; }
        public virtual Autoservis? Autoservis { get; set; }
        public int? ZaposlenikId { get; set; }
        public virtual Zaposlenik? Zaposlenik { get; set; }
        public int? Kolicina { get; set; }
        public decimal? UkupnaCijenaProizvoda { get; set; }
    }
}

File: CarCareHub_/UI_/flutter_desktop/lib/models/korpa.dart
import 'package:flutter_mobile/models/autoservis.dart';
import 'package:flutter_mobile/models/klijent.dart';
import 'package:flutter_mobile/models/product.dart';
import 'package:flutter_mobile/models/zaposlenik.dart';
import 'package:json_annotation/json_annotation.dart';

part 'korpa.g.dart';

@JsonSerializable()
class Korpa {
  int? korpaId;
  int? proizvodId;
  Product? proizvod;
  int? klijentId;
  Klijent? klijent;
  int? autoservisId;
  Autoservis? autoservis;
  int? zaposlenikId;
  Zaposlenik? zaposlenik;
  int? kolicina;
  double? ukupnaCijenaProizvoda;
  bool? vidljivo;

  Korpa({
    this.korpaId,
    this.vidljivo,
    this.proizvodId,
    this.klijentId,
    this.autoservisId,
    this.zaposlenikId,
    this.kolicina,
    this.ukupnaCijenaProizvoda,
  });

  factory Korpa.fromJson(Map<String, dynamic> json) => _$KorpaFromJson(json);

  Map<String, dynamic> toJson() => _$KorpaToJson(this);
}
